scripts/docs/extension_instructions.md

```diff
@@ -10,154 +10,91 @@ working with the same Bareflank workspace directory structure.
-    DEFAULT_VAL Option1
-    OPTIONS Option1 Option2 Option3
-    DESCRIPTION "An example selectable-string configuration"
-)
-
-# Add a boolean configuration that only shows up in cmake-gui as "advanced"
-add_config(
-    CONFIG_NAME EXAMPLE_BOOL_CONFIG
-    CONFIG_TYPE BOOL
-    DEFAULT_VAL ON
-    DESCRIPTION "An example boolean configuration"
-    ADVANCED
-)
+cmake_minimum_required(VERSION 3.6)
+project(example_vmm C CXX)
 
-# Add a path configuration
-add_config(
-    CONFIG_NAME EXAMPLE_PATH_CONFIG
-    CONFIG_TYPE PATH
-    DEFAULT_VAL "/the/default/value/for/this/path"
-    DESCRIPTION "An example path configuration"
+include(${SOURCE_CMAKE_DIR}/project.cmake)
+init_project(
+    INCLUDES ${CMAKE_CURRENT_LIST_DIR}/../include
 )
 
-# Add a file configuration that doesn't get validated by the build system
-add_config(
-    CONFIG_NAME EXAMPLE_PATH_CONFIG
-    CONFIG_TYPE PATH
-    DEFAULT_VAL "/the/default/path/to/this/file.txt"
-    DESCRIPTION "An example file configuration"
-    SKIP_VALIDATION
+add_vmm_executable(example_vmm
+    SOURCES example_vcpu_factory.cpp
 )
 ```
 
-Then, the build configuration file (bfconfig.cmake) can specify values for these
-configs as follows:
-
-```
-vmm_extension(
-    my_extension
-    SOURCE_DIR ../my_extension
-    DEPENDS extended_apis
-    CMAKE_ARGS
-        -DEXAMPLE_BOOL_CONFIG=OFF
-        -DEXAMPLE_PATH_CONFIG=/a/path/different/from/the/deafult
-)
-```
+The first 4 instructions are mandatory before adding any targets via
+`add_vmm_executable`, `add_static_library` or `add_shared_library`.
 
-Additionally, the base Bareflank build system provides some build configurations
-to all Bareflank extensions by default. To view and manipulate all
-extension-specific build configurations, use ccmake or cmake-gui:
+And finally, build your extension from within your build directory:
 
 ```
-ccmake ~/bareflank/build/extensions/<extension_name>/build
-```
-
-## Extension Build Rules
-
-Bareflank extensions can also provide extension-specific build validation rules
-using the same mechanism as the base hypervisor. If any of these build rules
-are violated, the build system will error before it begins building the
-extension. To declare extension-specific build rules, add a file named
-build_rules.cmake to the top level directory of the extension, and use the
-*add_build_rule* macro. For example:
-
+cmake ../hypervisor
+make
 ```
-# You can use build configs from the base hypervisor:
-add_build_rule(
-    FAIL_ON ${BUILD_TARGET_ARCH} NOT STREQUAL x86_64
-    FAIL_MSG "This extension is only supported on x86_64"
-)
 
-# Or you can use extension-specific build configs:
-add_build_rule(
-    FAIL_ON ${EXAMPLE_OPTION_CONFIG} STREQUAL Option3 AND NOT ${EXAMPLE_BOOL_CONFIG}
-    FAIL_MSG "Cannot build with EXAMPLE_OPTION_CONFIG Option3 while EXAMPLE_BOOL_CONFIG is OFF"
-)
-```
+For a complete example, look at the
+[hypervisor_exmaple_cpuidcount](https://github.com/Bareflank/hypervisor_example_cpuidcount/tree/dev)
```

Comment:
As part of this, please provide a PR for this example that merges dev into master so that we can provide a real link here